lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -838,36 +840,70 @@ class ADContext {
   SILPassManager &getPassManager() const { return passManager; }
   Lowering::TypeConverter &getTypeConverter() { return module.Types; }
 
-  SmallVectorImpl<DifferentiableFunctionInst *> &
-  getDifferentiableFunctionInsts() {
-    return differentiableFunctionInsts;
+  /// Returns true if the `differentiable_function` instruction worklist is
+  /// empty.
+  bool isDifferentiableFunctionInstsWorklistEmpty() const {
+    return differentiableFunctionInsts.empty();
   }
 
-  SmallPtrSetImpl<DifferentiableFunctionInst *> &
-  getProcessedDifferentiableFunctionInsts() {
-    return processedDifferentiableFunctionInsts;
+  /// Pops and returns a `differentiable_function` instruction from the
+  /// worklist. Returns nullptr if the worklist is empty.
+  DifferentiableFunctionInst* popDifferentiableFunctionInstFromWorklist() {
+    if (differentiableFunctionInsts.empty()) return nullptr;
+    return differentiableFunctionInsts.pop_back_val();
   }
 
-  llvm::SmallMapVector<SILDifferentiabilityWitness *, DifferentiationInvoker,
-                       32> &
-  getInvokers() {
+  /// Adds the given `differentiable_function` instruction to the worklist.
+  void addDifferentiableFunctionInst(DifferentiableFunctionInst* dfi) {
```

Comment:
minor, for consistency:
```suggestion
  void addDifferentiableFunctionInst(DifferentiableFunctionInst *dfi) {
```
Could you please update the rest of the added helper functions taking pointer-typed arguments?

Reply:
Hmm...I thought I did a `clang-format`. Anyway, I will fix them.

Reply:
By default, `clang-format` seems to leave `*` and `&` positions untouched. There are configuration fields like `PointerAlignment` though: https://clang.llvm.org/docs/ClangFormatStyleOptions.html